README.md

```diff
@@ -78,6 +77,32 @@ An async `for` loop will propagate errors out of the function so `message` has
 the `Item` type of the `stream` passed in. Note that async `for` loops can only
 be used inside of an `#[async]` function.
 
+And finally, you can create a `Stream` instead of a `Future` via
+`#[async_stream(item = _)]`:
+
+```rust
+#[async]
+fn fetch(client: hyper::Client, url: &'static str) -> io::Result<String> {
+    // ...
+}
+
+/// Fetch all provided urls one at a time
+#[async_stream(item = String)]
+fn fetch_all(client: hyper::Client, urls: Vec<&'static str>) -> io::Result<()> {
+    for url in urls {
+        let s = await!(fetch(client, url))?;
+        stream_yield!(Ok(s));
```

Comment:
Was this intended to be `stream_yield!(s)`?

Reply:
Ah so it looks like this was changed during discussion?

I'd personally prefer to only yield the item type from `stream_yield!`. Our eventual desired semantics for `Stream` is that an `Err` fuses (ends) the stream as well as returning `None`, so I'm hoping we can get a jump start on that here by only allowing yielding items and using `return Err(...)` for errors.

Reply:
Awesome, that's a change I'm very much in favour of. For some reason I got the impression from the linked issue that that wouldn't be changing any time soon, if it is a concrete plan then I can definitely switch this back.